### DISCUSSION TITLE
The reaktion force via th_to_csv th_to_csv does not show the reaction force.

### DISCUSSION THREAD

**Discussion by SonSiont** [OpenRadioss/OpenRadioss#867] [Q&A]
**Describe the bug**
The reaktion force via th_to_csv th_to_csv does not show the reaction force.

**To Reproduce**
Run reakForceIssue_0000.rad, reakForceIssue_0001.rad and th_to_csv_linux64_gf.
In Paraview/Editor you can see the wrong curve.

**Expected behavior**
The reaction force should be displayed.

**Output**
An archive with .rad, .out, .T01, .csv and some png files.
You can also add the `*.rad` ,`*.out` files.

**Desktop (please complete the following information):**
    Git ID 6684c24be8ff686a570509c4b835a555a865c3fb
- see the .out files in the attached archive.

**Additional context**
In the model, some nodes of a component are moved through a surface and pulled back.
The force should be zero afterwards and visible in the requested diagram. The contact pressure goes correct back to zero.
[reakForceIssue.tar.gz](https://github.com/OpenRadioss/OpenRadioss/files/10122030/reakForceIssue.tar.gz)

**Comment by sebastienVilleneuve** [comment]
@SonSiont 
The curve that you see is the impulse one 
( No forces are outputed in T01 file , only impulses)

To be able to get the force value , converter will derivate the impulse for you automatically , but for this , you need to add /TH/TITLE in 1.rad file before running computation ( cf https://github.com/OpenRadioss/OpenRadioss/tree/main/tools/th_to_csv)

![image](https://user-images.githubusercontent.com/100404030/204772345-7f5c6e48-d96f-48db-b7d1-50bb55ea84c1.png)

I will add some note here

Regards,
Sébastien

**Comment by SonSiont** [comment]
Thanks, it works

**Comment by sebastienVilleneuve** [comment]
Corrected by : https://github.com/OpenRadioss/OpenRadioss/pull/866

**Comment by rzadki22** [comment]
Hello @sebastienVilleneuve,

I am writing here to not create redundant threads. I probably found an issue with the th_to_csv converter. 

Let me introduce my benchmark: I had ran 3 jobs with a different /TFILE parameter: 0.01, 0.05 and 0.1.

TH outputs from 0.1 differs much in comparison to others because it already includes reaction derivatives (by time). This inconsistency sources probably from some text parsing issue in the th_to_csv exec. 

Lets consider variable called in my example the "TH_Measuring Nodes" on the node 3218  (it is main node of the RBODY) and variable REACX.* Depending on the /TFILE parameter the corresponding headline in the csv file has different blank character in the end:

0.01 --> 'TH_Measuring_Nodes                       3218                                           REACX     \t '
0.05 --> 'TH_Measuring_Nodes                       3218                                           REACX     \x01 '
0.1  --> 'TH_Measuring_Nodes                       3218                                           REACX      '

As you can see only spacebar character at the end provide expected behaviour of the converter so I suspect parsing issue as the source of this confusing conversion.

In the attachment you can find my examples and pictures. I enclosed raw plot of column with reaction and its derivative by time. I am using rather python for postprocessing than excel and this is why I noticed above-mentioned difference : )

I also found that .vtk files show expected values of reaxction as independent on ANIM save step.

*I noticed similar behaviour with /TH/RBODY/1 and a variable FX

I am looking forward for information about improvement of converter : )

Best regards
Filip


Attachment:
[csv_his.zip](https://github.com/OpenRadioss/OpenRadioss/files/13653839/csv_his.zip)
